src/utils/EventEmitterBasedNonNullState.ts

```diff
@@ -0,0 +1,15 @@
+import EventEmitterBasedState from "./EventEmitterBasedState";
+
+export default class EventEmitterBasedNonNullState<
```

Comment:
why do you need this wrapper? in what case null state is valid?

Reply:
Mostly for typescript. Null is the default initial state. But this wrapper is for when you an actual initial state. By the current state design, the value is only null able before initialisation after that the state will never change back to null. And the listeners are only notified on when the non null value has changed.